### PR TITLE
feat: allow download the accounting rules

### PR DESCRIPTION
**Pull request by lukicenturi**
## Checklist

- [x] Allow exporting accounting rules from response JSON, when not passing the `directory_path`
- [x] Fix the PATCH API for importing accounting rules, when `async_query` is `true`

discord thread https://discord.com/channels/657906918408585217/1245296485411717151

### CODE REVIEW

File: rotkehlchen/api/v1/resources.py [hunk 1]
```diff
@@ -282,6 +283,17 @@ def _combine_parser_data(
...
 
+def _maybe_get_temp_file(filepath: FileStorage | Path) -> Path:
+    """Checks if filepath is a FileStorage and if so,
+    creates a temporary copied file and returns its path."""
+    if isinstance(filepath, FileStorage):
+        _, tmpfilepath = tempfile.mkstemp()
+        filepath.save(tmpfilepath)
+        return Path(tmpfilepath)
+
+    return filepath
```
Comment (yabirgb) [inline rotkehlchen/api/v1/resources.py:294]: Shouldn't this happen in `AsyncFilePathSchema`. This function can be a staticmethod in that class

File: rotkehlchen/tests/api/test_accounting_rules.py [hunk 2]
```diff
@@ -530,25 +532,47 @@ def test_cache_invalidation(rotkehlchen_api_server: APIServer):
...
+                'accountingrulesexportresource',
+            ), json={'async_query': async_query},
+        )
+        if async_query:
+            response_result = wait_for_async_task_with_result(
+                server=rotkehlchen_api_server,
+                task_id=response.json()['result']['task_id'],
+            )
+        else:
+            response_result = assert_proper_response_with_result(response)
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_accounting_rules.py:555]: have you checked if this structure is something common around the code? Maybe we could have a `check_async_request` that does this. YOu are repeting code

File: rotkehlchen/tests/api/test_makerdao_dsr.py [hunk 3]
```diff
@@ -462,11 +460,11 @@ def test_query_current_dsr_balance(
             rotkehlchen_api_server,
             'makerdaodsrbalanceresource',
         ), json={'async_query': async_query})
-        if async_query:
-            task_id = assert_ok_async_response(response)
-            outcome = wait_for_async_task_with_result(rotkehlchen_api_server, task_id)
-        else:
-            outcome = assert_proper_response_with_result(response)
+        outcome = assert_maybe_async_response_with_result(
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_makerdao_dsr.py:463]: Can U rename this to remove maybe_async from the name? As it's now an arg?

Comment (LefterisJP) [reply rotkehlchen/tests/api/test_makerdao_dsr.py]: Did that already. All good.

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**